pkg/virt-api/webhooks/arm64.go

```diff
@@ -0,0 +1,157 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+/*
+ * arm64 utilities are in the webhooks package because they are used both
+ * by validation and mutation webhooks.
+ */
+package webhooks
+
+import (
+	"fmt"
+	rt "runtime"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+var _false bool = false
+var VerifyArchFunc = defaultVerifyArchFunc
+
+const (
+	defaultCPUModel = "host-passthrough"
+)
+
+// verifyInvalidSetting verify if VMI spec contain unavailable setting for arm64, check following items:
+// 1. if setting bios boot
+// 2. if use uefi secure boot
+// 3. if set auto-attach graphics device
+// 4. if use host-model for cpu model
+func verifyInvalidSetting(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) (metav1.StatusCause, bool) {
+	if spec.Domain.Firmware != nil && spec.Domain.Firmware.Bootloader != nil {
+		if spec.Domain.Firmware.Bootloader.BIOS != nil {
+			return metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueNotSupported,
+				Message: "Arm64 does not support bios boot, please change to uefi boot",
+				Field:   field.Child("domain", "firmware", "bootloader", "bios").String(),
+			}, false
+		}
+		if spec.Domain.Firmware.Bootloader.EFI != nil {
+			// When EFI is enable, secureboot is enabled by default, so here check two condition
+			// 1 is EFI is enabled without Secureboot setting
+			// 2 is both EFI and Secureboot enabled
+			if spec.Domain.Firmware.Bootloader.EFI.SecureBoot == nil || (spec.Domain.Firmware.Bootloader.EFI.SecureBoot != nil && *spec.Domain.Firmware.Bootloader.EFI.SecureBoot) {
+				return metav1.StatusCause{
+					Type:    metav1.CauseTypeFieldValueNotSupported,
+					Message: "UEFI secure boot is currently not supported on aarch64 Arch",
+					Field:   field.Child("domain", "firmware", "bootloader", "efi", "secureboot").String(),
+				}, false
+			}
+		}
+	}
+	if &spec.Domain.Devices != nil && spec.Domain.Devices.AutoattachGraphicsDevice != nil && *spec.Domain.Devices.AutoattachGraphicsDevice {
```

Comment:
Maybe we should change the default for arm? We can do this in a followup. I am unsure. Is there another graphics device which we could attach by default?

Reply:
> Maybe we should change the default for arm? We can do this in a followup. I am unsure. Is there another graphics device which we could attach by default?

Ok, in case you miss the explanation on why I need disable autoattachgraphicsdevice, I put related issue here https://github.com/kubevirt/libvirt/issues/49
For display device, only following display device is supported in qemu-system-aarch64
I need give it a try to see how to use them.
```
Display devices:
name "virtio-gpu-device", bus virtio-bus
name "virtio-gpu-pci", bus PCI, alias "virtio-gpu"
```

Reply:
If e.g. `virtio-gpu-device` works, that would be nice. But we can deal with it in a follow-up PR.